src/Contract/Monad.purs

```diff
@@ -1,7 +1,9 @@
 -- | A module defining the `Contract` monad.
 module Contract.Monad
-  ( Contract(..)
```

Comment:
If we're going `derive instance Newtype (Contract r a) _`, we may as well export the data constructor, as we can just `wrap` and `unwrap`?
You should get a warning like
```
An export for Contract hides data constructors but the type declares an instance of Data.Newtype.Newtype.
    Such instance allows to match and construct values of this type, effectively making the constructors public.
  ```